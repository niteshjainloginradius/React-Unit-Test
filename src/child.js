import React from 'react';
export default class Child extends React.Component {
    constructor(props){
        super(props);
    }
    render() {
       return (
          <div id="child">
             {this.props.children}
          </div>
       );
    }
 }