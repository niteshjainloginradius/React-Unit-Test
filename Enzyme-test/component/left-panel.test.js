import React from 'react';
import jestMock from  'jest-mock';
import render from 'react-test-renderer'; 
import Enzyme, {shallow, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { createSerializer } from 'enzyme-to-json';
import {Form, Col, Button, Row } from 'react-bootstrap';
import LeftPanel from './../../src/left-panel';
import child  from './../../src/child';

// for snapshot testing requirement
createSerializer({mode: "deep"});
Enzyme.configure({adapter : new Adapter()});


// element render testing requirement

const email='nitesh.jain@loginradius.com';
const pass='123456';
const rememberMe=false;

describe('Step:1 Snapshot Testing.',()=>{
    
    const tree = render.create(<LeftPanel/>);
    const testInstance = tree.root;
    test('Unit Test Snapshot',done=>{
        expect(tree.toJSON()).toMatchSnapshot();// if match with previou one then true else false.
        //expect(testInstance.findByType(child).props.children).toBe(1);
        done()
    })
    test('Unit Test Snapshot',done=>{
        const tree1 = render.create('body');
        expect(tree1.toJSON()).toMatchSnapshot();// if match with previou one then true else false.
        //expect(testInstance.findByType(child).props.children).toBe(1);
        done()
    })
}) 
describe('Step:2 Element Are There On Component.',()=>{
    const wrapper= shallow(<LeftPanel/>);
    let leftPanel= new LeftPanel();
    test('Unit Test Component With HTML Element',done=>{
    //1. we need to check our componet render correctly.
    //https://airbnb.io/enzyme/docs/api/ShallowWrapper/containsMatchingElement.html
    expect(wrapper.find(child).length).toBe(1); // sub component check 
    expect(wrapper.containsMatchingElement(<Form.Control name="email" type="email" placeholder="Email" onChange={LeftPanel.changeText}/>)).toBe(true); 
    expect(wrapper.containsMatchingElement(<Form.Control name="password" placeholder="Password" onChange={LeftPanel.changeText}/>)).toBe(true);
    expect(wrapper.containsMatchingElement(<Form.Check name="remember" label="Remember me" value={leftPanel.state.remember} onChange={LeftPanel.rememberMe} />)).toBe(true);
    expect(wrapper.containsMatchingElement(<Button  type="button" onClick={LeftPanel.myEvent}>Sign in</Button>)).toBe(true);
    done();
    })
}) 

 // component mount testing
describe('Step:3 Event Mock.',()=>{
    test('simulates click events',()=>{
        const onButtonClick = jestMock.fn();
        let button= shallow(<Button controlId="buttonSign" type="button" onClick={onButtonClick}>Sign in</Button>)
        button.find('button').simulate('click');
        expect(onButtonClick.mock.calls.length).toBe(1);

    })
})

describe('Step: 4 Dom Testing.',()=>{
       test('Submit Button Click Event Data Collecting.',()=>{
         const wrapperMount= mount(<LeftPanel/>,{attachTo: document.getElementById('root')});
         wrapperMount.find('#formHorizontalEmail').simulate('change', {target: {value: email, name: 'email'}});
         wrapperMount.find('#formHorizontalPassword').simulate('change', {target: {value: pass, name:'password'}});
         wrapperMount.find('#formHorizontalCheck').simulate('change', {target: {value: true, name:'remember'}});
         wrapperMount.find('button').simulate('click');
         expect(wrapperMount.find('#child').text()).toBe('{"email":"nitesh.jain@loginradius.com","password":"123456","remember":true,"childTextMessage":"ghjgjgjgj"}')
         wrapperMount.unmount();
       })
})

