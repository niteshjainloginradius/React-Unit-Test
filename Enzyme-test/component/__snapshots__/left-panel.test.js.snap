// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step:1 Snapshot Testing. Unit Test Snapshot 1`] = `
<div>
  <form
    className=""
  >
    <div
      className="form-group row"
    >
      <label
        className="form-label col-form-label col-sm-2"
      >
        Email
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="form-control"
          id="formHorizontalEmail"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
        />
      </div>
    </div>
    <div
      className="form-group row"
    >
      <label
        className="form-label col-form-label col-sm-2"
      >
        Password
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="form-control"
          id="formHorizontalPassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
    </div>
    <div
      className="form-group row"
    >
      <div
        className="col-sm-10 offset-sm-2"
      >
        <div
          className="form-check"
        >
          <input
            className="form-check-input"
            disabled={false}
            id="formHorizontalCheck"
            name="remember"
            onChange={[Function]}
            type="checkbox"
            value={false}
          />
          <label
            className="form-check-label"
            htmlFor="formHorizontalCheck"
            title=""
            type="checkbox"
          >
            Remember me
          </label>
        </div>
      </div>
    </div>
    <div
      className="form-group row"
    >
      <div
        className="col-sm-10 offset-sm-2"
      >
        <button
          className="btn btn-primary"
          controlId="buttonSign"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Sign in
        </button>
      </div>
    </div>
  </form>
  <div
    id="childrean"
  >
    <div
      id="child"
    >
      
    </div>
  </div>
</div>
`;

exports[`Step:1 Snapshot Testing. Unit Test Snapshot 2`] = `"body"`;
